shared/route-tree/index.js

```diff
@@ -28,7 +28,7 @@ type RouteDefParams<P> = {
   defaultSelected?: string,
   tags?: LeafTags,
   initialState?: {},
-  children: {},
+  children: {} | (name: string) => RouteDefNode,
```

Comment:
It's actually `{[key: string]: RouteDefParams<P> | () => RouteDefNode} | (name: string) => RouteDefNode`. Updated!